nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -577,6 +669,20 @@ def forward(
                 else:
                     attention_mask = attention_mask[..., : attention_scores.size(3), : attention_scores.size(3)]
 
+        if position_bias is None:
+            if self.position_embedding_type == 'relative':
+                position_bias = self.compute_bias(real_seq_length, key_length)
+            else:
+                pass # HuggingFace implementation initialize position_bias to zero when not using
+
+            # if key and values are already calculated
+            # we want only the last query position bias
+            if layer_past is not None:
+                position_bias = position_bias[:, :, -hidden_states.size(0) :, :]
+
+            position_bias = position_bias + attention_mask
+            attention_scores += position_bias
```

Comment:
if `position_bias` is None and `self.position_embedding_type != 'relative'`, 
this will break  since you are doing `position_bias = position_bias + attention_mask` and position_bias is None